annotation_pipeline/image.py

```diff
@@ -7,35 +7,96 @@
 
 from annotation_pipeline.utils import ds_dims, get_pixel_indices
 from annotation_pipeline.validate import make_compute_image_metrics, formula_image_metrics
+from annotation_pipeline.segment import ISOTOPIC_PEAK_N
+
+
+class ImagesManager:
+    max_formula_images_size = 512 * 1024 ** 2  # 512MB
+
+    def __init__(self, ibm_cos, bucket, prefix):
+        self.formula_metrics = {}
+        self.formula_images = {}
+
+        self._formula_images_size = 0
+        self._ibm_cos = ibm_cos
+        self._bucket = bucket
+        self._prefix = prefix
+        self._partition = 0
+
+    def __call__(self, f_i, f_metrics, f_images):
+        self.add_f_metrics(f_i, f_metrics)
+        self.add_f_images(f_i, f_images)
+
+    @staticmethod
+    def images_size(f_images):
+        return sum(img.data.nbytes + img.row.nbytes + img.col.nbytes for img in f_images if img is not None)
+
+    def add_f_images(self, f_i, f_images):
+        self.formula_images[f_i] = f_images
+        self._formula_images_size += ImagesManager.images_size(f_images)
+        if self._formula_images_size > self.__class__.max_formula_images_size:
+            self.save_images()
+            self.formula_images.clear()
+            self._formula_images_size = 0
+
+    def add_f_metrics(self, f_i, f_metrics):
+        self.formula_metrics[f_i] = f_metrics
+
+    def save_images(self):
+        if self.formula_images:
+            print(f'Saving {len(self.formula_images)} images')
+            self._ibm_cos.put_object(Bucket=self._bucket,
+                                     Key=f'{self._prefix}/{self._partition}.pickle',
```

Comment:
Please update `Pipeline.get_images` with this new key format

Reply:
@LachlanStuart I added support for `get_images()` using the new `cloudobject` class. PyWren can produce these objects to save and download from COS without the necessity of providing keys or prefixes.
pay attention that if you use `data_cleaner: True` in the config file, all images will be deleted, so I suggest to not use `data_cleaner: True` anymore, that's why I gathered all PyWren executors into one member variable of the `Pipeline` class - so we can use `self.pywren_executor.clean()` to clean all temporary data.

Reply:
> @LachlanStuart I added support for get_images() using the new cloudobject class. 

Very nice!

> I gathered all PyWren executors into one member variable of the Pipeline class - so we can use self.pywren_executor.clean() to clean all temporary data.

Good call. This makes the code tidier too.